### ISSUE TITLE
VRMS-03 checkIn route in client is broken

### ISSUE THREAD

**Issue by akibrhast** [hackforla/VRMS#601]
### Overview
- Home page is broken
- Check in system is broken

### More details to be added in the comments section

![image](https://user-images.githubusercontent.com/32349001/123519609-6e4fa680-d67a-11eb-96f7-f76ed6a61ec1.png)

### The files in question
- [CheckInfForm.js](https://github.com/hackforla/VRMS/blob/development/client/src/pages/CheckInForm.js)
- [NewUserForm.js](https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/components/presentational/newUserForm.js)
- [ReturnUserForm.js](https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/components/presentational/returnUserForm.js)

### The routes in question
- http://localhost:3000/checkIn/newUser?eventId=5eb9e2de3409ffef8194ffde
- http://localhost:3000/checkIn/returningUser?eventId=5eb9e2de3409ffef8194ffde
- http://localhost:3000/newProfile

**Comment by akibrhast** [comment]
### Some notes/details

- In regards to `/client/CheckInForm.js` which is routed via `{ path: '/checkIn/:userType', name: 'checkIn', Component: CheckInForm },` 
on the `/client/App.js` file

- The function `submitNewProfile` is not being used at all. It is being called in `createNewProfile` which is not being called at all

- Not sure what nesseciates the function `checkEmail` , the email check is already being done during the creation of the user. Which disallows non-unique emails.

- What is the difference between these two routes in `/client/App.js`
```js
  { path: '/checkIn/:userType', name: 'checkIn', Component: CheckInForm },
  { path: '/newProfile', name: 'newProfile', Component: CheckInForm },
```

---

- What is the purpose behind the function `handleReasonChange` https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/pages/CheckInForm.js#L114 . It seems to have been passed to the `ReturnUserForm` component as a prop,  https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/pages/CheckInForm.js#L616 . But in the component itself, it has been commented out code for a while now. https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/components/presentational/returnUserForm.js#L90

- The same problem as above can be seen across the following lines
    - https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/pages/CheckInForm.js#L36
    - https://github.com/hackforla/VRMS/blob/c269a30118ae436b8d8836cb005f831a54d5d455/client/src/pages/CheckInForm.js#L15
 
I believe there are many such instances of unused functions and props.

---
- the same code is repeated multiple times. I am talking about the requests sections. What happened to being DRY ? Can we refactor this into something more modular?

This piece of code and versions of it is copy pasted multiple times
```js
                return fetch('/api/checkins', {
                    method: "POST",
                    body: JSON.stringify(checkInForm),
                    headers: {
                        "Content-Type": "application/json",
                        "x-customrequired-header": headerToSend
                    }
                })
```
I am thinking we could make a generic function that goes along the lines of
```js
    static async request(endpoint, paramsOrData = {}, verb = "get") {
      // paramsOrData._token = ( // for now, hardcode token for "testing"
      // "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc" +
      // "3RpbmciLCJpc19hZG1pbiI6ZmFsc2UsImlhdCI6MTU1MzcwMzE1M30." +
      // "COmFETEsTxN_VfIlgIKw0bYJLkvbRQNgO1XCSE8NZ0U");
      paramsOrData._token = localStorage.getItem('_token');
      console.debug("API Call:", endpoint, paramsOrData, verb);
  
      try {
        return (await axios({
          method: verb,
          url: `http://localhost:3001/${endpoint}`,
          [verb === "get" ? "params" : "data"]: paramsOrData})).data;
          // axios sends query string data via the "params" key,
          // and request body data via the "data" key,
          // so the key we need depends on the HTTP verb
      }
  
      catch(err) {
        console.error("API Error:", err.response);
        let message = err.response.data.message;
        throw Array.isArray(message) ? message : [message];
      }
    }
  
```

---


### Resource

- https://github.com/akibrhast/Joblify/blob/637aca2793b32ff9c30e1948ec43d74086a369cf/frontend/app/src/services/JoblyApi.js#L6
-  An attempt to fix: https://github.com/hackforla/VRMS/pull/602
-  Related: https://github.com/hackforla/VRMS/issues/603

**Comment by akibrhast** [comment]
Based on the object  `question` it seems to me that the Current Role and Desired Role fields are supposed to be dropdowns, no?

![image](https://user-images.githubusercontent.com/32349001/123519446-56c3ee00-d679-11eb-8fc9-a5bacd829e07.png)

![image](https://user-images.githubusercontent.com/32349001/123519513-a86c7880-d679-11eb-8815-301c952e2cee.png)

**Comment by akibrhast** [comment]
### A WIP PR that is trying to address the problems
- https://github.com/hackforla/VRMS/pull/602

**Comment by akibrhast** [comment]
Deleted question about hack nights cause we dont have hack nights anymore

**Comment by akibrhast** [comment]
- Checkin model in backend allows for checkingIn for a NULL event. 
    - [ ] Investigate and note does all mongo shell command that will delete all existing checkIn entries with a NULL eventId
    - [ ] CheckIn model eventId needs to be set to become a required field

-  Checkin model in backend allows for checkingIn for a NULL user.
    - [ ] Investigate and note does all mongo shell command that will delete all existing checkIn entries with a NULL userId
    - [ ]  CheckIn model userI needs to be set to become a required field

**Comment by akibrhast** [comment]
Why use fetch and not axios?
 
- Yes fetch has come a long way, yes it's built into vanilla. Yes axios has a lot of dependencies. But Axios still provides an  easier handling for network requests when compared to fetch. 
- But we aren't trying to maintain v03 over any long period of time, its supposed to be a POC . From my understanding we are trying to get it to a functioning state at minimum.

**Comment by akibrhast** [comment]
~~- The api call on the home page, `/client/src/pages/Home.js` is broken. It is calling the endpoint functionality, 
`"/api/events?checkInReady=true"` which does not exist. Which causes the home page to NEVER show any events that might be available for checking in.
    - [ ] Fix this by possibly adding a new controller method, that queries the event entry for checkInReady=true events.~~
This is not a problem, as pointed out by alex anakin

**Comment by akibrhast** [comment]
I am not sure how much things I should move around in the backend, it is very inconvenient to figure out which sections of the code belongs to 03 and which ones for 04. This is probably a dumb question that I feel has possibly been answered already but why on earth are we not separating the 03 and 04 to their own branches. @alex-anakin  ?

**Comment by akibrhast** [comment]
@hackforla/vrms-write

**Comment by akibrhast** [comment]
- On successful check-in with a valid email as a returning user, the client redirects to the success page. However, when using the returning user endpoint to check in as random email. Nothing happens.
   -[ ] Add display notification on non-valid email for returning user check-in

**Comment by ExperimentsInHonesty** [comment]
Wont fix.  This will be resolved with the launch of v0.4 which wont use these file.  We currently believe that this is not causing the site to operate outside of expected. (based on the dev instance being able to be logged into currently).
